File: fm-dddrive/src/main/java/io/dddrive/core/validation/model/enums/CodeValidationLevelEnum.java
package io.dddrive.core.validation.model.enums;

import io.dddrive.core.enums.model.base.EnumerationBase;

public class CodeValidationLevelEnum extends EnumerationBase<CodeValidationLevel> {

	public static CodeValidationLevel INFO;
	public static CodeValidationLevel WARNING;
	public static CodeValidationLevel ERROR;

	private static CodeValidationLevelEnum INSTANCE;

	public CodeValidationLevelEnum() {
		super(CodeValidationLevel.class);
		INSTANCE = this;
		this.addItem(new CodeValidationLevel(this, "info", "Info"));
		this.addItem(new CodeValidationLevel(this, "warning", "Warning"));
		this.addItem(new CodeValidationLevel(this, "error", "Error"));
	}

	public static CodeValidationLevelEnum getInstance() {
		return INSTANCE;
	}

	public static CodeValidationLevel getValidationLevel(String validationLevelId) {
		return INSTANCE.getItem(validationLevelId);
	}

	@Override
	public void assignItems() {
		INFO = getValidationLevel("info");
		WARNING = getValidationLevel("warning");
		ERROR = getValidationLevel("error");
	}

}

File: fm-dddrive/src/main/java/io/dddrive/validation/model/enums/CodeValidationLevelEnum.kt
package io.dddrive.validation.model.enums

import io.dddrive.enums.model.base.EnumerationBase

class CodeValidationLevelEnum : EnumerationBase<CodeValidationLevel>(CodeValidationLevel::class.java) {

	init {
		instance = this
		this.addItem(CodeValidationLevel("info", "Info"))
		this.addItem(CodeValidationLevel("warning", "Warning"))
		this.addItem(CodeValidationLevel("error", "Error"))
	}

	override fun assignItems() {
		INFO = getValidationLevel("info")
		WARNING = getValidationLevel("warning")
		ERROR = getValidationLevel("error")
	}

	companion object {

		lateinit var INFO: CodeValidationLevel

		lateinit var WARNING: CodeValidationLevel

		lateinit var ERROR: CodeValidationLevel

		lateinit var instance: CodeValidationLevelEnum

		@JvmStatic
		fun getValidationLevel(validationLevelId: String): CodeValidationLevel = instance.getItem(validationLevelId)

	}

}
